sweep/sweeper.go

```diff
@@ -859,17 +957,16 @@ func (s *UtxoSweeper) sweep(inputs inputSet,
 		tx, labels.MakeLabel(labels.LabelTypeSweepTransaction, nil),
 	)
 	if err != nil {
+		// TODO(yy): find out which input is causing the failure.
+		s.markInputsPublishFailed(tx.TxIn)
```

Comment:
Given that failed is a terminal state, this can be dangerous if the error here is transient as I believe it will cause us to abort any retrying logic.

Reply:
Indeed. My plan is to do a binary search on the inputs here, assuming only one bad input, then we can keep searching by retrying tx creation and checking mempool acceptance, till we find the bad actor. Something to improve may be in 18.1

Reply:
Do we have an issue for this?

Reply:
not yet - think i'll do a summary of what I wanna do in an issue page once the final PR is reviewed, turns out there are looots of TODOs there🤦🏻